### ISSUE TITLE
Crash on AS OF NOW

### ISSUE THREAD

**Issue by krizhanovsky** [tempesta-tech/mariadb#397]
### Reproduce

I faced the crash only once, so I have no idea which steps exactly lead to the problem, but full steps are:
```sql
> create table t(x int primary key) with system versioning;
> insert into t values (1);
> alter table t drop system versioning;
> alter table t add column y int;
> alter table t add system versioning;                                                
> set @@versioning_alter_history=keep;
> alter table t modify y int without system versioning;
> update t set y=1;
> update t set x=1;
> select * from t for system_time as of now;
ERROR 2013 (HY000): Lost connection to MySQL server during query
```

### Calltrace

Trunk with latest commit cb4657e3b42e273318344181754a0cf8fb393524

```
Thread 33 "mysqld" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7ffff4158700 (LWP 10803)]
0x0000555555eda4cb in Item_field::type_handler (this=0x7fff5c013910)
    at /home/alex/proj/mariadb/mariadb/sql/item.h:2821
2821	    const Type_handler *handler= field->type_handler();
Missing separate debuginfos, use: dnf debuginfo-install libgcc-6.3.1-1.fc24.x86_64 libstdc++-6.3.1-1.fc24.x86_64 sssd-client-1.14.2-1.fc24.x86_64 zlib-1.2.8-10.fc24.x86_64
(gdb) bt
#0  0x0000555555eda4cb in Item_field::type_handler (this=0x7fff5c013910) at /home/alex/proj/mariadb/mariadb/sql/item.h:2821
#1  0x0000555555af985d in Item::result_type (this=0x7fff5c013910) at /home/alex/proj/mariadb/mariadb/sql/item.h:817
#2  0x0000555555ce5d21 in vers_select_conds_t::resolve_units (this=0x7fff5c014028, timestamps_only=false) at /home/alex/proj/mariadb/mariadb/sql/table.cc:8877
#3  0x0000555555c07796 in st_select_lex::vers_setup_conds (this=0x7fff5c004fd0, thd=0x7fff5c000b00, tables=0x7fff5c013a20, where_expr=0x7fff5c014550)
    at /home/alex/proj/mariadb/mariadb/sql/sql_select.cc:900
#4  0x0000555555c089ef in JOIN::prepare (this=0x7fff5c014148, tables_init=0x7fff5c013a20, wild_num=1, conds_init=0x0, og_num=0, order_init=0x0, skip_order_by=false, 
    group_init=0x0, having_init=0x0, proc_param_init=0x0, select_lex_arg=0x7fff5c004fd0, unit_arg=0x7fff5c004870) at /home/alex/proj/mariadb/mariadb/sql/sql_select.cc:1104
#5  0x0000555555c13a9e in mysql_select (thd=0x7fff5c000b00, tables=0x7fff5c013a20, wild_num=1, fields=..., conds=0x0, og_num=0, order=0x0, group=0x0, having=0x0, 
    proc_param=0x0, select_options=2147748608, result=0x7fff5c014128, unit=0x7fff5c004870, select_lex=0x7fff5c004fd0)
    at /home/alex/proj/mariadb/mariadb/sql/sql_select.cc:4171
#6  0x0000555555c05dee in handle_select (thd=0x7fff5c000b00, lex=0x7fff5c0047a8, result=0x7fff5c014128, setup_tables_done_option=0)
    at /home/alex/proj/mariadb/mariadb/sql/sql_select.cc:368
#7  0x0000555555bd0f3b in execute_sqlcom_select (thd=0x7fff5c000b00, all_tables=0x7fff5c013a20) at /home/alex/proj/mariadb/mariadb/sql/sql_parse.cc:6529
#8  0x0000555555bc7162 in mysql_execute_command (thd=0x7fff5c000b00) at /home/alex/proj/mariadb/mariadb/sql/sql_parse.cc:3754
#9  0x0000555555bd48a0 in mysql_parse (thd=0x7fff5c000b00, rawbuf=0x7fff5c0136f8 "select * from t for system_time as of now", length=41, parser_state=0x7ffff4157640, 
    is_com_multi=false, is_next_command=false) at /home/alex/proj/mariadb/mariadb/sql/sql_parse.cc:7985
#10 0x0000555555bc1b41 in dispatch_command (command=COM_QUERY, thd=0x7fff5c000b00, packet=0x7fff5c00b061 "select * from t for system_time as of now", packet_length=41, 
    is_com_multi=false, is_next_command=false) at /home/alex/proj/mariadb/mariadb/sql/sql_parse.cc:1829
#11 0x0000555555bc04dd in do_command (thd=0x7fff5c000b00) at /home/alex/proj/mariadb/mariadb/sql/sql_parse.cc:1370
#12 0x0000555555d25d2d in do_handle_one_connection (connect=0x555558763130) at /home/alex/proj/mariadb/mariadb/sql/sql_connect.cc:1420
#13 0x0000555555d25a80 in handle_one_connection (arg=0x555558763130) at /home/alex/proj/mariadb/mariadb/sql/sql_connect.cc:1326
#14 0x00007ffff7bc45ca in start_thread () from /lib64/libpthread.so.0
#15 0x00007ffff6a05f6d in clone () from /lib64/libc.so.6
```

**Comment by midenok** [comment]
Duplicate of #398.
